packages/shared/src/__tests__/telemetry.test.ts

```diff
@@ -0,0 +1,89 @@
+import { TelemetryCollector } from '../telemetry';
+
+jest.useFakeTimers();
+
+describe('TelemetryCollector', () => {
```

Comment:
Added some high-level tests for the collector. Most importantly, validating that events aren't sent when `disabled` or `debug` are true.